src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessWaitState.Unix.cs

```diff
@@ -278,12 +277,11 @@ internal ManualResetEvent EnsureExitedEvent()
                         if (!_isChild)
                         {
                             // If we haven't exited, we need to spin up an asynchronous operation that
-                            // will completed the exitedEvent when the other process exits. If there's already
-                            // another operation underway, then we'll just tack ours onto the end of it.
-                            _waitInProgress = _waitInProgress == null ?
-                                WaitForExitAsync() :
-                                _waitInProgress.ContinueWith((_, state) => ((ProcessWaitState)state!).WaitForExitAsync(),
-                                    this, CancellationToken.None, TaskContinuationOptions.None, TaskScheduler.Default).Unwrap();
+                            // will complete the _exitedEvent when the other process exits. If there's already
+                            // another operation underway, then we'll just tack ours onto the end of it; we can't
+                            // be sure it'll actually monitor the process until it exits, as it may have been
+                            // created with a cancelable token.
+                            _waitInProgress = WaitForExitAsync(_waitInProgress, CancellationToken.None);
```

Comment:
Could this be:
```
_waitInProgress ??= WaitForExitAsync(...)
```

Reply:
If _waitInProgress completing guaranteed the process had exited, then yes.  But as noted, that's not the case.  And as such this code, which does need a task that will complete only when the process exits, needs to tack on its own, just in case.